clients/src/main/java/org/apache/kafka/common/requests/LeaderAndIsrRequest.java

```diff
@@ -35,25 +36,6 @@
 
 public class LeaderAndIsrRequest extends AbstractRequest {
 
-    public static class PartitionState {
```

Comment:
I think the code "duplication" here is okay, since these are 2 completely separate requests that should be able to evolve independently. 

I know there was discussion that the 2 requests may be a bit redundant (I don't remember the details now)...but thats a separate patch/discussion.